mitxgraders/helpers/calc/expressions.py

```diff
@@ -528,7 +528,7 @@ def parse(self, expression):
             return self.cache[cache_key]
 
         try:
-            parsed = self.raw_parse(expression)
+            parsed = self.raw_parse(expression_no_whitespace)
```

Comment:
expression_no_whitespace  was defined and we just forgot to use it? Good catch :-)